lib/refile/s3.rb

```diff
@@ -60,12 +67,14 @@ def initialize(region:, bucket:, max_size: nil, prefix: nil, hasher: Refile::Ran
     verify_uploadable def upload(uploadable)
       id = @hasher.hash(uploadable)
 
-      if uploadable.is_a?(Refile::File) and uploadable.backend.is_a?(S3) and uploadable.backend.access_key_id == access_key_id
-        object(id).copy_from(copy_source: [@bucket_name, uploadable.backend.object(uploadable.id).key].join("/"))
+      operation, options = if upload_via_copy_operation?(uploadable)
+        [:copy_from, { copy_source: [@bucket_name, uploadable.backend.object(uploadable.id).key].join("/") }]
       else
-        object(id).put(body: uploadable, content_length: uploadable.size)
+        [:put, { body: uploadable, content_length: uploadable.size }]
       end
 
+      object(id).send(operation, s3_options_for(operation, options))
```

Comment:
In the previous version with AWS SDK V 1.x it was possible to pass these options as defaults to the client object whereas now they are only available as options for the "operation" methods (`copy_from`, `put`, `presigned_post`).

`presigned_post` and `copy_from/put` have slightly different APIs regarding their options (as described in #3). The two options implemented here happen to share same option keys across the three operations, but for example options to provide custom encryption keys seem to have different hash keys in `presigned_post` than in other two.

So the indirection is there so that it'll be simple to add the other keys from the respective operation APIs when someone needs them. If supporting those options is over-architecting at this point then a simpler implementation is indeed possible.